tests/unit/getAccountIds.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable operator-linebreak */
+'use strict'; // eslint-disable-line
+
+const assert = require('assert');
+const { errors } = require('arsenal');
+const http = require('http');
+const querystring = require('querystring');
+const IAMClient = require('../../lib/IAMClient.js');
+
+const canId1 =
+    '0123456789012345678901234567890123456789012345678901234567890123';
+const canId2 =
+    '1234567890123456789012345678901234567890123456789012345678901230';
+const canId3 =
+    '2345678901234567890123456789012345678901234567890123456789012301';
+const canIdBad =
+    '2345678901234567890123456789012345678901234567890123456789012bad';
+
+const accountId1 = 'accountId1';
+const accountId2 = 'accountId2';
+const accountId3 = 'accountId3';
+
+const serverDB = {};
+serverDB[accountId1] = canId1;
+serverDB[accountId2] = canId2;
+serverDB[accountId3] = canId3;
+
+serverDB[canId1] = accountId1;
+serverDB[canId2] = accountId2;
+serverDB[canId3] = accountId3;
+
+function handler(req, res) {
+    const index = req.url.indexOf('?');
+    const data = querystring.parse(req.url.substring(index + 1));
+    let inputArray = data.canonicalIds;
+    if (!Array.isArray(inputArray)) {
+        inputArray = [inputArray];
+    }
+    if (inputArray.indexOf(canIdBad) !== -1) {
+        res.writeHead(errors.InvalidParameterValue.code);
+        return res.end(JSON.stringify(errors.InvalidParameterValue));
+    }
+    const output = [];
+    inputArray.forEach(canId => output.push({
+        id: serverDB[canId],
+        canId,
+    }));
```

Comment:
forEach + push can be map

```suggestion
    const output = inputArray.map(canId => {
        return {
            id: serverDB[canId],
            canId,
        }
    });
```